runtime/compiler/net/CommunicationStream.hpp

```diff
@@ -112,6 +113,10 @@ class CommunicationStream
             int bytesRead = (*OBIO_read)(_ssl, data + totalBytesRead, size - totalBytesRead);
             if (bytesRead <= 0)
                {
+               if ((*OBIO_should_retry)(_ssl))
+                  {
+                  JITServerHelpers::setRetryConnectionImmediately(true);
```

Comment:
I am a bit weary about the concurrency issues due to this code. The code in `postStreamFailure` is protected by a critical section, but here we set flags outside the critical section. I can't even think of all the possible combination of actions.

Reply:
This "retry connection immediately" flag seems to be per-connection rather then global, so I think it should be stored in the stream object.

Reply:
It could also be reported in the `StreamFailure` itself and passed in to `postStreamFailure`.